coap/messages.go

```diff
@@ -0,0 +1,155 @@
+package adapter
+
+import (
+	"bytes"
+	"log"
+	"net"
+
+	mux "github.com/dereulenspiegel/coap-mux"
+	coap "github.com/dustin/go-coap"
+	"github.com/mainflux/mainflux/writer"
+)
+
+// sendMessage sends the message to NATS
+func (ca *CoAPAdapter) sendMessage(l *net.UDPConn, a *net.UDPAddr, m *coap.Message) *coap.Message {
+	log.Printf("Got message in sendMessage: path=%q: %#v from %v", m.Path(), m, a)
+	var res *coap.Message
+	if m.IsConfirmable() {
+		res = &coap.Message{
+			Type:      coap.Acknowledgement,
+			Code:      coap.Content,
+			MessageID: m.MessageID,
+			Token:     m.Token,
+			Payload:   []byte(""),
+		}
+		res.SetOption(coap.ContentFormat, coap.AppJSON)
+	}
+
+	if len(m.Payload) == 0 {
+		if m.IsConfirmable() {
+			res.Payload = []byte("{\"res\": \"Error: msg len can not be 0\"}")
+		}
+		return res
+	}
+
+	// Channel ID
+	cid := mux.Var(m, "channel_id")
+
+	// Publish message via NATS
+	n := writer.RawMessage{}
+	n.Channel = cid
+	n.Publisher = ""
+	n.Protocol = "coap"
+	n.Payload = m.Payload
+
+	if err := ca.repo.Save(n); err != nil {
+		if m.IsConfirmable() {
+			res.Code = coap.InternalServerError
+		}
+		return res
+	}
+
+	if m.IsConfirmable() {
+		res.Code = coap.Changed
+	}
+	return res
+}
+
+// registerObserver functions adds observer struct to the observers map
```

Comment:
Private methods does not require godoc. On top of this, function is named properly, hence, this doc only clutters the code.